plugins/hydra_joblib_launcher/hydra_plugins/hydra_joblib_launcher/joblib_launcher.py

```diff
@@ -76,20 +76,25 @@ def launch(self, job_overrides: Sequence[Sequence[str]]) -> Sequence[JobReturn]:
         configure_log(self.config.hydra.hydra_logging, self.config.hydra.verbose)
         sweep_dir = Path(str(self.config.hydra.sweep.dir))
         sweep_dir.mkdir(parents=True, exist_ok=True)
+
+        # Joblib's backend is hard-coded to loky since the threading
+        # backend is incompatible with Hydra's logging
+        joblib_keywords = OmegaConf.to_container(self.joblib).copy()
```

Comment:
1. Why are you converting to primitive dict?
2. no need to copy, to_container is already copying.
3. use resolve=True to ensure any interpolation is resolved.

Reply:
1. I want to "backend" to the arguments for `Joblib.Parallel` without having it in the config (to hard-code it). If I did `self.joblib["backend"] = "loky"`, I'd get: `AttributeError: Accessing unknown key in a struct : hydra.joblib.backend`

Will fix 2. and 3.

Reply:
Gotcha.
Read [this](https://omegaconf.readthedocs.io/en/latest/usage.html#configuration-flags) for more context on why you get it.
converting to a container is one way to deal with it, another is to open the structure to changes with open_dict.

I actually think having backend: "loki" hard coded in the config and raising an exception if the user try to set it to threads is better.
1. I will prevent threads from being used.
2. I will make it easy for others to try a different backend.
it will most likely fail but at least they can try to make with config changes only.
We can documented that the only tested and backend is loki

Reply:
I can easily do that, however, this will be the situation:

- The `backend` keyword of Joblib 0.14.1 allow selection between `['loky', 'multiprocessing', 'sequential', 'threading']`:
  - `multiprocessing` won't work without major plugin changes. It fails when trying to pickle the task function and config loader. Beyond that, Joblib advises using `loky` rather than `multiprocessing` in the docs, since it's considered more stable
  - `sequential` enables sequential computation
  - `threading` is incompatible
- In effect, `loky` is the only reasonable choice for `backend` at the moment
- Alternative backends like `dask`, `hadoop`, or `ray` (!), are not selected through the `backend` keyword of `Joblib.Parallel` but use context managers and require additional configuration and plugin changes

What do you think?

Reply:
Okay - it sounds like Loki is the only sane choice right now, so your solution sounds good.